cli/src/test/java/org/jboss/gm/cli/DifferentJVMTest.java

```diff
@@ -49,38 +58,67 @@
+     * This location ($HOME/.gradle/jdks) was chosen to match https://docs.gradle.org/current/userguide/toolchains.html.
      * It utilises the same directory structure and location thereby matching potential prior downloads.
      */
     @BeforeClass
-    public static void setupJVM() {
-        if (SystemUtils.IS_OS_LINUX && !jdk8.exists()) {
-            //noinspection ResultOfMethodCallIgnored
-            gradleJDKHome.mkdirs();
-            File tarFile = new File(gradleJDKHome + File.separator + jdkTar);
-            File compressedFile = new File(tarFile + ".gz");
+    public static void setupJVM() throws IOException {
+        assumeTrue(SystemUtils.IS_OS_LINUX);
+
+        String filename = null;
+        UnArchiver ua = null;
+
+        if (SystemUtils.IS_OS_LINUX) {
+            filename = "OpenJDK8U-jdk_x64_linux_hotspot_8u272b10.tar.gz";
+            ua = new TarGZipUnArchiver();
+        } else if (SystemUtils.IS_OS_WINDOWS) {
+            filename = "OpenJDK8U-jdk_x64_windows_hotspot_8u272b10.zip";
+            ua = new ZipUnArchiver();
+        } else {
+            assumeTrue(filename != null && ua != null);
+        }
+
+        if (!Files.exists(JDK8_BIN_DIR)) {
+            System.out.println("JDK8_DIR: " + JDK8_DIR);
+            Files.createDirectories(JDK8_DIR);
+
+            Path destFile = GRADLE_JDK_HOME.resolve(filename);
+            System.out.println("destFile: " + destFile);
+
             Unirest.get(
-                    "https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u272-b10/OpenJDK8U-jdk_x64_linux_hotspot_8u272b10.tar.gz")
-                    .asFile(compressedFile.toString());
+                    "https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u272-b10/" + filename)
+                    .asFile(destFile.toString());
+
+            System.out.println("sourceFile: " + destFile.toFile() + ", destDirectory: " + GRADLE_JDK_HOME.toFile());
+            ((LogEnabled) ua).enableLogging(new ConsoleLoggerManager().getLoggerForComponent("plexus-archiver"));
+            ua.setSourceFile(destFile.toFile());
+            ua.setDestDirectory(GRADLE_JDK_HOME.toFile());
+            System.out.println("Extract!");
+
+            assertThat(destFile).isRegularFile();
+            assertThat(GRADLE_JDK_HOME).isDirectory();
 
-            final TarGZipUnArchiver ua = new TarGZipUnArchiver();
-            ua.enableLogging(new ConsoleLoggerManager().getLoggerForComponent("plexus-archiver"));
-            ua.setSourceFile(compressedFile);
-            ua.setDestDirectory(tarFile.getParentFile());
             ua.extract();
 
-            compressedFile.deleteOnExit();
+            System.out.println("Set hook!");
+            Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+                try {
+                    Files.deleteIfExists(destFile);
+                } catch (IOException e) {
+
+                }
+            }));
```

Comment:
>> @rnc 
>> Why the change to use shutdown hook?

> @dwalluck 
> Only because I switched to Path which didn't have that API directly, but I can change this back.

Yeah I think so especially as we are only running on Linux